### DISCUSSION TITLE
WPF as netstandard2.0 target

### DISCUSSION THREAD

**Discussion by bstordrup** [dotnet/wpf#9651] [Ideas]
If I would like to build a library with some WPF functionality that should be avilable on both `net462`+ and `net6`+ I have to dual target the library.

But would it make sense to be able to target the library as `netstandard2.0`? I mean, `netstandard2.0` is least common denominator for `net462` and `netcoreapp3.1`, and by targeting it to `netstandard2.0`, you would have a package that would go directly into every target framework, that supports `netstandard2.0`.

**Comment by miloush** [comment]
.NET Standard does not include any WPF API. If you need to use dependency properties/dispatcher/elements etc. you need to dual target.

**Reply by bstordrup** [reply]
But it also limits the usage to projects that either dual target the same targets or any of them. You cannot use the project in another non-wpf `netstandard2.0` project.

My scenario is having a dispatcher library that takes some types from WPF (HwndSource) for passing forward. I need to reference this library from a `netstandard2.0` project, but I cannot because the dispatcher library has to target `net462` and/or `netcoreapp3.1`.

Might be that the solution to this problem would be a bit more complex by removing the actual WPF references and break them down to more regular primitives and pass these instead to he place where they are actually needed.

**Reply by miloush** [reply]
Your understanding is correct, but that seems to be by design. .NET Standard projects cannot use WPF types because they are not available in .NET Standard, and hence to a project targeting .NET Standard, neither directly nor through a library.

If you need the types internally only, you might be able to include the necessary types directly in your library. If you need to consume or produce them, then I guess reflection is the only option, but obviously that breaks the .NET Standard contract.

.NET Standard is only a list of API available, it's not a runtime. If your project is targeting `netstandard2.0`, it can run on any platform, including e.g. embedded processors, as long as it can provide that list of API. It's not guaranteed to have anything else like HwndSource.
